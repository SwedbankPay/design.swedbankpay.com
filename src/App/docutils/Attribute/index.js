import React from "react";
import PropTypes from "prop-types";

export const Property = ({ value }) => <code className="token property">{value}</code>;

const Attribute = ({ data, name, value }) => {
    if (name && value) {
        return (
            <code>
                {data ? <span className="token attr-name">data-</span> : null}
                <span className="token attr-name">{name}</span>
                <span className="token attr-value">
                    <span className="token punctuation">{"=\""}</span>
                    {value}
                    <span className="token punctuation">{"\""}</span>
                </span>
            </code>
        );
    } else if (name && !value) {
        return (
            <code>
                {data ? <span className="token attr-name">data-</span> : null}
                <span className="token attr-name">{name}</span>
            </code>
        );
    } else if (!name && value) {
        return (
            <code>
                <span className="token attr-value">{value}</span>
            </code>
        );
    }
};

Attribute.propTypes = {
    name: PropTypes.string,
    value: PropTypes.string,
    data: PropTypes.bool
};

Property.propTypes = {
    value: PropTypes.string.isRequired
};

export default Attribute;
