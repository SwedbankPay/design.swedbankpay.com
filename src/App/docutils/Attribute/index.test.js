import React from "react";
import { shallow } from "enzyme";
import "prismjs";

import Attribute, { Property } from "./index";

describe("Utilities: Attribute", () => {
    it("is defined", () => {
        expect(Attribute).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Attribute name="test" value="testValue" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<span className="token attr-name">test</span>)).toEqual(true);
    });

    it("renders data attribute", () => {
        const wrapper = shallow(<Attribute name="test" value="test" data />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("data");
    });

    it("renders name when only the name prop has been passed", () => {
        const wrapper = shallow(<Attribute name="test" />);
        const spans = wrapper.find("span");

        expect(wrapper).toMatchSnapshot();
        expect(spans).toHaveLength(1);
        expect(wrapper.html()).not.toContain("data");
    });

    it("renders name with data when the name and data prop has been passed", () => {
        const wrapper = shallow(<Attribute name="test" data />);
        const spans = wrapper.find("span");

        expect(wrapper).toMatchSnapshot();
        expect(spans).toHaveLength(2);
        expect(wrapper.html()).toContain("data");
    });

    it("renders value when only the value prop has been passed", () => {
        const wrapper = shallow(<Attribute value="test" />);
        const spans = wrapper.find("span");

        expect(wrapper).toMatchSnapshot();
        expect(spans).toHaveLength(1);
        expect(wrapper.contains(<span className="token attr-value">test</span>)).toEqual(true);
    });

    it("renders only value when value and data has been passed", () => {
        const wrapper = shallow(<Attribute value="test" data />);
        const spans = wrapper.find("span");

        expect(wrapper).toMatchSnapshot();
        expect(spans).toHaveLength(1);
        expect(wrapper.html()).not.toContain("data");
        expect(wrapper.contains(<span className="token attr-value">test</span>)).toEqual(true);
    });
});

describe("Utilities: property", () => {
    it("renders with value prop", () => {
        const wrapper = shallow(<Property value="test"/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<code className="token property">test</code>)).toEqual(true);
    });
});
