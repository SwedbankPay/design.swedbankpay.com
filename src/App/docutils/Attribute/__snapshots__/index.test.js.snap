// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: Attribute renders 1`] = `
<code>
  <span
    className="token attr-name"
  >
    test
  </span>
  <span
    className="token attr-value"
  >
    <span
      className="token punctuation"
    >
      ="
    </span>
    testValue
    <span
      className="token punctuation"
    >
      "
    </span>
  </span>
</code>
`;

exports[`Utilities: Attribute renders data attribute 1`] = `
<code>
  <span
    className="token attr-name"
  >
    data-
  </span>
  <span
    className="token attr-name"
  >
    test
  </span>
  <span
    className="token attr-value"
  >
    <span
      className="token punctuation"
    >
      ="
    </span>
    test
    <span
      className="token punctuation"
    >
      "
    </span>
  </span>
</code>
`;

exports[`Utilities: Attribute renders name when only the name prop has been passed 1`] = `
<code>
  <span
    className="token attr-name"
  >
    test
  </span>
</code>
`;

exports[`Utilities: Attribute renders name with data when the name and data prop has been passed 1`] = `
<code>
  <span
    className="token attr-name"
  >
    data-
  </span>
  <span
    className="token attr-name"
  >
    test
  </span>
</code>
`;

exports[`Utilities: Attribute renders only value when value and data has been passed 1`] = `
<code>
  <span
    className="token attr-value"
  >
    test
  </span>
</code>
`;

exports[`Utilities: Attribute renders value when only the value prop has been passed 1`] = `
<code>
  <span
    className="token attr-value"
  >
    test
  </span>
</code>
`;

exports[`Utilities: property renders with value prop 1`] = `
<code
  className="token property"
>
  test
</code>
`;
